components/playlist/resources/reducers/playlist_reducer.ts

```diff
@@ -12,14 +12,27 @@ import * as PlaylistMojo from 'gen/brave/components/playlist/mojom/playlist.mojo
 const playlistReducer: Reducer<Playlist.State|undefined> =
     (state: Playlist.State|undefined, action) => {
       if (state === undefined) {
-        state = { lists: [] }
+        state = { lists: [], currentList: undefined }
       }
 
       // TODO(sko) Handle each action properly.
       switch (action.type) {
         case types.PLAYLIST_LOADED:
-          state = { ...state, lists: [...action.payload as PlaylistMojo.Playlist[]] }
+          const playlists = action.payload as PlaylistMojo.Playlist[]
```

Comment:
Is the type here not inferred? I think it should be.

Reply:
Aaah, maybe it's because we haven't specified a type for action?

```ts
(state: Playlist.State | undefined, action: PlayListLoadedAction | PlaylistSelectedAction) => {
    switch (action.type) {
        case types.PLAYLIST_LOADED:
            // Stuff
            break;
        case types.PLAYLIST_SELECTED:
            // More stufff
            break;
    }
}
```

Reply:
> Is the type here not inferred? I think it should be.

Yes, The narrowing works. It was just for me not to be confused. Removed :)